platform-sdk/swirlds-logging/src/test/java/com/swirlds/logging/util/LoggingTestScenario.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.logging.util;
+
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.extensions.test.fixtures.TestConfigBuilder;
+import com.swirlds.logging.api.Level;
+import com.swirlds.logging.api.internal.level.HandlerLoggingLevelConfig;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * {@link LoggingTestScenario} is a relationship between a desired configuration and the list of assertions to verify
+ * for that config
+ */
+public final class LoggingTestScenario {
+    private final String scenarioName;
+    private final TestConfigBuilder scenarioConfigBuilder;
+    private final List<AssertionRule> assertionsRules;
+
+    private LoggingTestScenario(String scenarioName, TestConfigBuilder builder, List<AssertionRule> assertionsRules) {
+        this.scenarioName = scenarioName;
+        this.scenarioConfigBuilder = builder;
+        this.assertionsRules = assertionsRules;
+    }
+
+    public Configuration configuration() {
+        return this.scenarioConfigBuilder.getOrCreateConfig();
+    }
+
+    public static TestScenarioBuilder builder() {
+        return new TestScenarioBuilder();
+    }
+
+    public static class TestScenarioBuilder {
```

Comment:
```suggestion
    public static class LoggingTestScenarioBuilder {
```

Reply:
The references are too much to suggest each one, and it could bring issues if applied, better to rename the whole class